pydantic/typing.py

```diff
@@ -258,6 +260,18 @@ def all_literal_values(type_: Type[Any]) -> Tuple[Any, ...]:
     return tuple(x for value in values for x in all_literal_values(value))
 
 
+def is_named_tuple_type(type_: Type[Any]) -> bool:
+    from .utils import lenient_issubclass
+
+    return lenient_issubclass(type_, tuple) and hasattr(type_, '_fields')
+
+
+def is_typed_dict_type(type_: Type[Any]) -> bool:
+    from .utils import lenient_issubclass
+
+    return lenient_issubclass(type_, dict) and getattr(type_, '__annotations__', None)
```

Comment:
Humm, I wonder if this is a water tight check. I can't think of a better solution though.

Reply:
Currently there is no easier way afaik. There will be an [public method](https://docs.python.org/3.10/library/typing.html#typing.is_typeddict) in 3.10 though